main.cpp

```diff
@@ -794,13 +798,34 @@ int main(int argc, char** argv)
     if (runAschedule)
         sched.Execute(utxo, txo);
     else
-    {
-        printf("Generate a block <enter>\n");
-        string input;
-        cin >> input;
         MaxSpeed(gc.bitcoind, utxo, txo);
+
+    if (!saveUtxos(txo))
+        printf("Could not save the remaining utxos");
+}
+
+bool saveUtxos(std::vector<UTXO> utxos)
+{
+    ofstream utxofile;
+    UniValue results(UniValue::VARR);
+    for (UTXO &utxo : utxos)
+    {
+        UniValue entry(UniValue::VOBJ);
+        entry.pushKV("txid", utxo.prevout.hash.GetHex());
+        entry.pushKV("vout", (uint64_t) utxo.prevout.n);
+        entry.pushKV("satoshi", utxo.satoshi);
+        CScript script = utxo.constraintScript;
+        entry.pushKV("scriptPubKey", HexStr(script.begin(), script.end()));
+        entry.pushKV("privKey", CBitcoinSecret(utxo.privKey).ToString());
+        results.push_back(entry);
```

Comment:
write the address as well (yeah I know that txunami doesn't need it)... but it could be useful for scripts that read this and do things like access the spends in an explorer.

Reply:
Yeah, I thought about it because it would be useful, but it's not part of the UTXO object, so didn't bother. 

I have now added the address.